src/java/org/apache/cassandra/service/accord/txn/TxnUpdate.java

```diff
@@ -77,13 +92,325 @@
+                }
+                else
+                {
+                    out.writeUnsignedVInt32(0);
+                }
+                serializeArrayUnsignedVInt(t.fragments, out);
+            }
+
+            @Override
+            public ConditionalBlock deserialize(DataInputPlus in) throws IOException
+            {
+                int id = in.readUnsignedVInt32();
+                ByteBuffer conditionBytes = readWithVIntLength(in);
+                SerializedTxnCondition condition = conditionBytes.remaining() > 0
+                                                   ? new SerializedTxnCondition(conditionBytes)
+                                                   : null;
+
+                // Deserialize mutations
+                int[] mutations = deserializeArrayUnsignedVInt(in);
+                return new ConditionalBlock(id, condition, mutations);
+            }
+
+            @Override
+            public void skip(DataInputPlus in) throws IOException
+            {
+                in.readUnsignedVInt32();
+                skipWithVIntLength(in);
+                skipArrayUnsignedVInt(in);
+            }
+
+            @Override
+            public long serializedSize(ConditionalBlock t)
+            {
+                long size = TypeSizes.sizeofUnsignedVInt(t.id);
+                if (t.condition != null)
+                {
+                    size += serializedSizeWithVIntLength(t.condition.bytes());
+                }
+                else
+                {
+                    size += TypeSizes.sizeofUnsignedVInt(0);
+                }
+                size += serializedArrayUnsignedVIntSize(t.fragments);
+                return size;
+            }
+        };
+        final int id;
+        /**
+         * If null, this block always executes (ELSE / unconditional).
+         */
+        @Nullable
+        final SerializedTxnCondition condition;
+        final int[] fragments;
+
+        ConditionalBlock(int id, @Nullable SerializedTxnCondition condition, int[] fragments)
+        {
+            this.id = id;
+            this.condition = condition;
+            this.fragments = fragments;
+        }
+
+        public long estimatedSizeOnHeap()
+        {
+            long size = 0; //TODO (correctness): EMPTY_SIZE
+            if (condition != null)
+                size += condition.estimatedSizeOnHeap();
+            size += ObjectSizes.sizeOfArray(fragments);
+            return size;
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (o == null || getClass() != o.getClass()) return false;
+            ConditionalBlock that = (ConditionalBlock) o;
+            return id == that.id && Objects.equals(condition, that.condition) && Arrays.equals(fragments, that.fragments);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(id, condition, Arrays.hashCode(fragments));
```

Comment:
do we use this hashCode, or can we throw `UnsupportedOperationException`? If not, can we just use `id`?